src/routes/labs-route.js

```diff
@@ -0,0 +1,70 @@
+const config = require('config');
+const multer = require('multer');
+const GridFsStorage = require('multer-gridfs-storage');
+const path = require('path');
+const crypto = require('crypto');
+const express = require('express');
+
+const storage = new GridFsStorage({
+  url: config.mongo,
+  connectionOpts: { useNewUrlParser: true },
+  file: (req, file) => {
+    return new Promise((resolve, reject) => {
+      crypto.randomBytes(16, (err, buf) => {
+        if (err) {
+          return reject(err);
+        }
+        const fileName = buf.toString('hex') + path.extname(file.originalname);
+
+        const fileInfo = {
+          filename: fileName,
+          bucket: 'uploads',
+        };
+        return resolve(fileInfo);
+      });
+    });
+  },
+});
+
+const upload = multer({ storage });
+
+module.exports = function createLabsRoute(repository) {
+  const router = express.Router();
+
+  router.route('/')
+    .get(list)
+    .post(upload.single('lab'), add);
+
+  router.route('/:filename')
+    .get(view);
+
+  router.route('/:id')
+    .delete(remove);
+
```

Comment:
why for view you use 'filename', but for delete 'id'

Reply:
You search by file name and you delete by id.